svm/src/transaction_processor.rs

```diff
@@ -377,14 +388,20 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
             account_keys_in_batch,
         );
 
-        let (mut validate_fees_us, mut load_us, mut execution_us): (u64, u64, u64) = (0, 0, 0);
+        // Create the transaction balance collector if recording is enabled.
+        let mut balance_collector = config
+            .recording_config
+            .enable_transaction_balance_recording
+            .then(|| BalanceCollector::new_with_transaction_count(sanitized_txs.len()));
+
+        let (mut load_us, mut execution_us): (u64, u64) = (0, 0);
 
```

Comment:
its only tangentially related to this pr, but instead of adding all timings at the bottom, i switched to adding as many timings as possible immediately after they are generated, and keeping `load_us` and `execution_us` as incremental accumulators because theyre printed in a debug message

this is because with `collect_balances_us`  we would have had four `mut u64` here, and if the program cache fix (filter account owners) ever lands we would have _seven_. it doesnt make much sense to keep this pattern, since we are just saturating-adding to a stack variable that is then later saturating-added to the timing struct anyway